tests/jest/IKUpload.test.jsx

```diff
@@ -205,6 +355,78 @@ describe('IKUpload', () => {
         expect(onChange.calledOnce).toEqual(true);
       });
+      test('should call onUploadStart when file input changes', () => {
+        // mount component
+        const ikUpload = mount(
+          <IKContext publicKey={publicKey} urlEndpoint={urlEndpoint} authenticationEndpoint={authenticationEndpoint} >
+            <IKUpload onError={onError} onSuccess={onSuccess} onUploadStart={onUploadStart} />
+          </IKContext>
+        );
+        // verify setup integrity
+        expect(ikUpload.html()).toEqual('<input type="file">');
+        expect(onUploadStart.called).toEqual(false);
+
+        // trigger file change and upload 
+        ikUpload.find('IKUpload').simulate('change', sampleEvent);
+
+        // verify change callback
+        expect(onUploadStart.calledOnce).toEqual(true);
+      });
+
+      test('should not upload file if validateFile fails', () => {
+        // mount component
+        const ikUpload = mount(
+          <IKContext publicKey={publicKey} urlEndpoint={urlEndpoint} authenticationEndpoint={authenticationEndpoint} >
+            <IKUpload onError={onError} onSuccess={onSuccess} validateFile={() => false} />
+          </IKContext>
+        );
+        // verify setup integrity
+        expect(ikUpload.html()).toEqual('<input type="file">');
+        expect(ikInstanceUploadStub.called).toEqual(false);
+
+        // trigger file change and upload 
+        ikUpload.find('IKUpload').simulate('change', sampleEvent);
+
+        // verify change callback
+        expect(ikInstanceUploadStub.called).toEqual(false);
+      });
+
+      test('should call onUploadProgress during file upload', () => {
+        // mount component
+        const ikUpload = mount(
+          <IKContext publicKey={publicKey} urlEndpoint={urlEndpoint} authenticationEndpoint={authenticationEndpoint} >
+            <IKUpload onError={onError} onSuccess={onSuccess} onUploadProgress={onUploadProgress} />
+          </IKContext>
+        );
+        // verify setup integrity
+        expect(ikUpload.html()).toEqual('<input type="file">');
+        expect(onUploadProgress.called).toEqual(false);
+
+        // trigger file change and upload 
+        ikUpload.find('IKUpload').simulate('change', sampleEvent);
+        expect(ikInstanceUploadStub.calledOnce).toEqual(true);
+
+        // verify change callback
+        expect(onUploadProgress.called).toEqual(true);
+      });
+
+      test('should call abort on xhr', () => {
+        // mount component
+        const ref = React.createRef();
+        const ikUpload = mount(
+          <IKContext publicKey={publicKey} urlEndpoint={urlEndpoint} authenticationEndpoint={authenticationEndpoint} >
+            <IKUpload onError={onError} onSuccess={onSuccess} ref={ref} />
+          </IKContext>
+        );
+        // verify setup integrity
+        expect(ikUpload.html()).toEqual('<input type="file">');
+        expect(ikUpload.find(IKUpload).getElement().ref).toBe(ref);
+        ref.current.abort();
```

Comment:
Aborting upload is not mentioned in readme. Also does it work like this? Should we test this in Cypress too?

Reply:
Have added details in readme. It works this way yes. As discussed, in cypress, we're not covering corner cases.